moveit_core/trajectory_processing/src/iterative_spline_parameterization.cpp

```diff
@@ -0,0 +1,576 @@
+    }
+
+    // Error out if bounds don't make sense
+    if (t2[j].max_velocity <= 0.0 || t2[j].max_acceleration <= 0.0)
+    {
+      logError("Joint %d max velocity %f and max acceleration %f must be greater than zero or a solution won't be "
+               "found.\n",
+               j, t2[j].max_velocity, t2[j].max_acceleration);
+      return false;
+    }
+    if (t2[j].min_velocity >= 0.0 || t2[j].min_acceleration >= 0.0)
+    {
+      logError("Joint %d min velocity %f and min acceleration %f must be less than zero or a solution won't be "
+               "found.\n",
+               j, t2[j].min_velocity, t2[j].min_acceleration);
+      return false;
+    }
+    if (jerk_enabled_ && t2[j].max_jerk <= 0.0)
+    {
+      logError("Joint %d max jerk %f must be greater than zero or a solution won't be found.\n", j, t2[j].max_jerk);
+      return false;
+    }
+    if (jerk_enabled_ && t2[j].min_jerk >= 0.0)
+    {
+      logError("Joint %d min jerk %f must be greater than zero or a solution won't be found.\n", j, t2[j].min_jerk);
+      return false;
+    }
+  }
+
+  for (unsigned int i = 0; i < num_points; i++)
+  {
+    robot_state::RobotStatePtr waypoint = trajectory.getWayPointPtr(i);
+    for (unsigned int j = 0; j < num_joints; j++)
+    {
+      t2[j].positions[i] = waypoint->getVariablePosition(idx[j]);
+      t2[j].velocities[i] = waypoint->getVariableVelocity(idx[j]);
+      t2[j].accelerations[i] = waypoint->getVariableAcceleration(idx[j]);
+    }
+  }
+
+  // Error check
+  if (time_change_factor_ <= 1.0)
+  {
+    logError("time change factor is %f, needs to be higher than 1.0.\n", time_change_factor_);
+    return false;
+  }
+  if (num_points < 4)
+  {
+    logError("number of waypoints %d, needs to be greater than 3.\n", num_points);
+    return false;
+  }
+  for (unsigned int j = 0; j < num_joints; j++)
+  {
+    if (t2[j].velocities[0] > t2[j].max_velocity || t2[j].velocities[0] < t2[j].min_velocity)
+    {
+      logError("Initial velocity %f out of bounds\n", t2[j].velocities[0]);
+      return false;
+    }
+    else if (t2[j].velocities[num_points - 1] > t2[j].max_velocity ||
+             t2[j].velocities[num_points - 1] < t2[j].min_velocity)
+    {
+      logError("Final velocity %f out of bounds\n", t2[j].velocities[num_points - 1]);
+      return false;
+    }
+    else if (t2[j].accelerations[0] > t2[j].max_acceleration || t2[j].accelerations[0] < t2[j].min_acceleration)
+    {
+      logError("Initial acceleration %f out of bounds\n", t2[j].accelerations[0]);
+      return false;
+    }
+    else if (t2[j].accelerations[num_points - 1] > t2[j].max_acceleration ||
+             t2[j].accelerations[num_points - 1] < t2[j].min_acceleration)
+    {
+      logError("Final acceleration %f out of bounds\n", t2[j].accelerations[num_points - 1]);
+      return false;
+    }
+  }
+
+  // Initialize times
+  // 0.01 to prevent divide-by-zero
+  std::vector<double> time_diff(trajectory.getWayPointCount() - 1, 0.01);
```

Comment:
Setting dt to some arbitrarily small value happens twice, here and at line [498](https://github.com/ros-planning/moveit/pull/441/files#diff-7bfd5dc84e8e4fae3ce118b47c59c3ffR498).  Is this necessary?

Also, I would propose considering a smaller value.  Depending on the controller update rate and trajectory density I think 0.01 could be too large for some cases.  What about something like `std::numeric_limits<double>::epsilon()` here instead?